lightning/src/lib.rs

```diff
@@ -27,10 +27,14 @@
 // that don't do anything but annoy us and cant actually ever be resolved.
 #![allow(bare_trait_objects)]
 #![allow(ellipsis_inclusive_range_patterns)]
+// Needed to silence warninge about #[macro_use] for alloc::vec,
+// remove when no_std is fully enabled
+#![allow(unused_attributes)]
```

Comment:
Hmm, I kinda hate doing this library-wide, is it possible to do it everywhere its needed? eg
```
#[allow(unused_attributes)] // In std the macro vec is already imported
#[macro_use]
use alloc::vec;
```

Reply:
How about the following.  It's more concise, and seems to eliminate the warnings.

at top level, in `lib.rs`:
```rust
mod prelude {
	pub use alloc::{vec, vec::Vec};
}
```

and everywhere else:
```rust
use prelude::*
```

Reply:
Tried the `prelude` approach, but got a ton of warnings. Maybe I did something wrong, but I ended up just removing the `#[macro_use]` from the `alloc::vec` imports.

Kind of weird, because I thought you had to include `#[macro_use]` for macro imports...

Anyway, appears to work now.

Reply:
I tried the prelude approach in https://github.com/lightning-signer/rust-lightning/tree/alloc and got no warnings.  Maybe grab that commit if you like it?

Reply:
Thanks, I forgot to `pub use`, your commit fixes that.